atomic_reactor/plugins/pre_add_dockerfile.py

```diff
@@ -67,45 +69,47 @@ def __init__(self, workflow, nvr=None, destdir=IMAGE_BUILD_INFO_DIR,
         self.use_final_dockerfile = use_final_dockerfile
 
         if nvr is None:
-            labels = Labels(df_parser(self.workflow.df_path).labels)
-            try:
-                _, name = labels.get_name_and_value(Labels.LABEL_TYPE_NAME)
-                _, version = labels.get_name_and_value(Labels.LABEL_TYPE_VERSION)
-                _, release = labels.get_name_and_value(Labels.LABEL_TYPE_RELEASE)
-            except KeyError as exc:
-                raise ValueError(
-                    "You have to specify either nvr arg or name/version/release labels."
-                ) from exc
-            nvr = "{0}-{1}-{2}".format(name, version, release)
-            nvr = nvr.replace("/", "-")
-        self.df_name = '{0}-{1}'.format(DOCKERFILE_FILENAME, nvr)
-        self.df_dir = destdir
-        self.df_path = os.path.join(self.df_dir, self.df_name)
-
-        # we are not using final dockerfile, so let's copy current snapshot
-        if not self.use_final_dockerfile:
-            local_df_path = os.path.join(self.workflow.df_dir, self.df_name)
-            shutil.copy2(self.workflow.df_path, local_df_path)
-
-    def run(self):
-        """
-        run the plugin
-        """
-        dockerfile = df_parser(self.workflow.df_path, workflow=self.workflow)
+            nvr = self._nvr_from_dockerfile()
+
+        self.df_name = f"{DOCKERFILE_FILENAME}-{nvr}"
+        self.df_path = os.path.join(destdir, self.df_name)
```

Comment:
Pathlib? Or is it not worth the trouble here?

Reply:
I'd say no, this one represents the destination path inside the container. I considered changing this to `f"{destdir}/{self.df_name}"`, since it is not an os-dependent path at all, but I decided to keep it as is since we handle in-container paths as os paths in others places as well